<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SecurityPage</class>
 <widget class="QWidget" name="SecurityPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>657</width>
    <height>587</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="SecurityLevelWidget" name="m_pSecurityIssues" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
     </widget>
     <widget class="QScrollArea" name="scrollArea">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="sizeIncrement">
       <size>
        <width>0</width>
        <height>8</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>625</width>
         <height>1126</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_8">
        <item>
         <widget class="QGroupBox" name="lrcfg_srtpEnabled">
          <property name="title">
           <string>Encr&amp;ypt media streams (SRTP)</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="AutoComboBox" name="combo_security_STRP">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="lrcfg_srtpRtpFallback">
             <property name="text">
              <string>Fallback on RTP on SDES failure</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="lrcfg_tlsEnabled">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Encrypt negotiation (&amp;TLS)</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="0">
            <widget class="QLabel" name="label_tls_authority">
             <property name="text">
              <string>Authority certificate list</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0" colspan="6">
            <widget class="QFrame" name="frame_2">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>55</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>66</height>
              </size>
             </property>
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_7">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_10">
                <item>
                 <widget class="QLabel" name="label_3">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="m_pMoveCertL_2">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>Your certificates access are too permissive. Certificates should only be readable by the owner.</string>
                  </property>
                  <property name="wordWrap">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_11">
                <item>
                 <widget class="QPushButton" name="m_pFixCertPB">
                  <property name="text">
                   <string>Fix the permissions</string>
                  </property>
                  <property name="icon">
                   <iconset theme="dialog-ok">
                    <normaloff>.</normaloff>.</iconset>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="m_pCancelFix">
                  <property name="text">
                   <string>No thanks</string>
                  </property>
                  <property name="icon">
                   <iconset theme="dialog-cancel">
                    <normaloff>.</normaloff>.</iconset>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item row="3" column="0" colspan="6">
            <widget class="QFrame" name="frame">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>55</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>66</height>
              </size>
             </property>
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_5">
                <item>
                 <widget class="QLabel" name="label_2">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="m_pMoveCertL">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>A good security practice is to move the certificates to a common directory. SELinux recommend ~/.cert for this. Do you wish to move the certificate there?</string>
                  </property>
                  <property name="wordWrap">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_4">
                <item>
                 <widget class="QPushButton" name="m_pMoveCertPB">
                  <property name="text">
                   <string>Move to ~/.cert</string>
                  </property>
                  <property name="icon">
                   <iconset theme="dialog-ok">
                    <normaloff>.</normaloff>.</iconset>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="m_pCancelMove">
                  <property name="text">
                   <string>No thanks</string>
                  </property>
                  <property name="icon">
                   <iconset theme="dialog-cancel">
                    <normaloff>.</normaloff>.</iconset>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item row="9" column="0">
            <widget class="QLabel" name="label_tls_timeout">
             <property name="text">
              <string>Negotiation timeout</string>
             </property>
            </widget>
           </item>
           <item row="13" column="0" colspan="4">
            <widget class="QCheckBox" name="lrcfg_tlsVerifyServer">
             <property name="text">
              <string>Verify incoming certificates (server side)</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_tls_endpoint">
             <property name="text">
              <string>Public endpoint certificate</string>
             </property>
            </widget>
           </item>
           <item row="17" column="0">
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="7" column="3" colspan="3">
            <spacer name="horizontalSpacer_5">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="6" column="1" colspan="5">
            <widget class="QLineEdit" name="lrcfg_tlsPassword">
             <property name="echoMode">
              <enum>QLineEdit::Password</enum>
             </property>
             <property name="placeholderText">
              <string>Only necessary if the key have a password</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_tls_private_key">
             <property name="text">
              <string>Private key</string>
             </property>
            </widget>
           </item>
           <item row="8" column="1" colspan="5">
            <widget class="QLineEdit" name="lrcfg_tlsServerName">
             <property name="placeholderText">
              <string>Usually the SIP server hostname</string>
             </property>
            </widget>
           </item>
           <item row="0" column="5">
            <widget class="QToolButton" name="m_pViewCa">
             <property name="toolTip">
              <string>View the certificate authority</string>
             </property>
             <property name="text">
              <string>View</string>
             </property>
             <property name="icon">
              <iconset theme="view-list-text">
               <normaloff>../../../../../../../../../../../../.designer/backup</normaloff>../../../../../../../../../../../../.designer/backup</iconset>
             </property>
            </widget>
           </item>
           <item row="1" column="5">
            <widget class="QToolButton" name="m_pViewCert">
             <property name="toolTip">
              <string>View the certificate</string>
             </property>
             <property name="text">
              <string>View</string>
             </property>
             <property name="icon">
              <iconset theme="view-list-text">
               <normaloff>../../../../../../../../../../../../.designer/backup</normaloff>../../../../../../../../../../../../.designer/backup</iconset>
             </property>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="label_tls_private_key_password">
             <property name="text">
              <string>Private key password</string>
             </property>
            </widget>
           </item>
           <item row="15" column="4">
            <widget class="IssuesIcon" name="m_pReqTLS" native="true"/>
           </item>
           <item row="15" column="0" colspan="4">
            <widget class="QCheckBox" name="lrcfg_tlsRequireClientCertificate">
             <property name="text">
              <string>Require a certificate for incoming TLS connections</string>
             </property>
            </widget>
           </item>
           <item row="9" column="1">
            <widget class="QSpinBox" name="lrcfg_tlsNegotiationTimeoutSec">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>50</width>
               <height>0</height>
              </size>
             </property>
             <property name="suffix">
              <string> seconds</string>
             </property>
            </widget>
           </item>
           <item row="7" column="1" colspan="2">
            <widget class="AutoComboBox" name="combo_tls_method"/>
           </item>
           <item row="14" column="4">
            <widget class="IssuesIcon" name="m_pVerifyClient" native="true"/>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="label_tls_outgoing">
             <property name="text">
              <string>Outgoing TLS server name</string>
             </property>
            </widget>
           </item>
           <item row="14" column="0" colspan="4">
            <widget class="QCheckBox" name="lrcfg_tlsVerifyClient">
             <property name="text">
              <string>Verify answer certificates (client side)</string>
             </property>
            </widget>
           </item>
           <item row="13" column="4">
            <widget class="IssuesIcon" name="m_pVerifyServer" native="true"/>
           </item>
           <item row="9" column="2" colspan="4">
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="11" column="0" colspan="6">
            <widget class="QListView" name="m_pCipherModel"/>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="label_tls_method">
             <property name="text">
              <string>TLS protocol method</string>
             </property>
            </widget>
           </item>
           <item row="10" column="0">
            <widget class="QCheckBox" name="m_pDefaultCiphers">
             <property name="text">
              <string>Use default ciphers</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="16" column="0" colspan="6">
            <widget class="QTabWidget" name="tabWidget">
             <property name="currentIndex">
              <number>0</number>
             </property>
             <widget class="QWidget" name="tab">
              <attribute name="title">
               <string>Allowed certificates</string>
              </attribute>
              <layout class="QGridLayout" name="gridLayout">
               <item row="0" column="0">
                <widget class="ContextListView" name="lrcfg_allowedCertificatesModel"/>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="tab_2">
              <attribute name="title">
               <string>Banned certificates</string>
              </attribute>
              <layout class="QGridLayout" name="gridLayout_2">
               <item row="0" column="0">
                <widget class="ContextListView" name="lrcfg_bannedCertificatesModel"/>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="tab_3">
              <attribute name="title">
               <string>Known certificates</string>
              </attribute>
              <layout class="QGridLayout" name="gridLayout_3">
               <item row="0" column="0">
                <widget class="ContextListView" name="lrcfg_knownCertificateModel"/>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
           <item row="0" column="1" colspan="4">
            <widget class="FileSelector" name="lrcfg_tlsCaListCertificate" native="true"/>
           </item>
           <item row="1" column="1" colspan="4">
            <widget class="FileSelector" name="lrcfg_tlsCertificate" native="true"/>
           </item>
           <item row="2" column="1" colspan="5">
            <widget class="FileSelector" name="lrcfg_tlsPrivateKey" native="true"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>AutoComboBox</class>
   <extends>QComboBox</extends>
   <header>widgets/autocombobox.h</header>
  </customwidget>
  <customwidget>
   <class>SecurityLevelWidget</class>
   <extends>QWidget</extends>
   <header>widgets/securityprogress.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>IssuesIcon</class>
   <extends>QWidget</extends>
   <header>widgets/securityprogress.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ContextListView</class>
   <extends>QListView</extends>
   <header>widgets/contextlistview.h</header>
  </customwidget>
  <customwidget>
   <class>FileSelector</class>
   <extends>QWidget</extends>
   <header>widgets/fileselector.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>m_pDefaultCiphers</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_pCipherModel</receiver>
   <slot>setHidden(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>55</x>
     <y>604</y>
    </hint>
    <hint type="destinationlabel">
     <x>148</x>
     <y>802</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pCancelMove</sender>
   <signal>clicked()</signal>
   <receiver>frame</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>613</x>
     <y>367</y>
    </hint>
    <hint type="destinationlabel">
     <x>616</x>
     <y>370</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pCancelFix</sender>
   <signal>clicked()</signal>
   <receiver>frame_2</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>613</x>
     <y>439</y>
    </hint>
    <hint type="destinationlabel">
     <x>616</x>
     <y>442</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
